test/plugin/test_out_google_cloud.rb

```diff
@@ -187,6 +242,16 @@ def test_managed_vm_metadata_loading
 
   def test_gce_metadata_does_not_load_when_use_metadata_service_is_false
     Fluent::GoogleCloudOutput.any_instance.expects(:fetch_metadata).never
+    d = create_driver(NO_METADATA_SERVICE_CONFIG + CUSTOM_METADATA_CONFIG)
+    d.run
+    assert_equal CUSTOM_PROJECT_ID, d.instance.project_id
+    assert_equal CUSTOM_ZONE, d.instance.zone
+    assert_equal CUSTOM_VM_ID, d.instance.vm_id
+    assert_equal false, d.instance.running_on_managed_vm
+  end
+
+  def test_metadata_overrides_on_gce
+    setup_gce_metadata_stubs
     d = create_driver(CUSTOM_METADATA_CONFIG)
     d.run
     assert_equal CUSTOM_PROJECT_ID, d.instance.project_id
```

Comment:
done.  I was debating whether to get rid of the CONFIG constants and inline the configuration in the create_driver call instead, which may make it more obvious.  I think the constants made more sense when there were only two of them but now that I have a bunch I don't think they're helping readability... so perhaps I will do that in a followup.